src/signature.c

```diff
@@ -833,10 +833,12 @@ static CK_RV p11prov_sig_operate_init(P11PROV_SIG_CTX *sigctx, bool digest_op,
         slotid = p11prov_obj_get_slotid(sigctx->key);
 
         ret = mech_fallback_init(sigctx, slotid);
+        goto done;
```

Comment:
Is this one correct?
mech_fallback_init() is a valid case, why would we skip login?
Maybe this branch needs to retry to grab a session instead (or cause mech_fallback_init() to return the session it allocates ?